orttraining/orttraining/core/graph/adasum_optimizer_graph_builder.cc

```diff
@@ -93,19 +68,87 @@ Status AdasumOptimizerGraphBuilder::BuildOptimizerNode(
       opt_configs, graph_defs,
       new_initializers,
       output_weight_argdefs, output_gradient_argdefs,
-      // Always enable grad clipping for Adasum
-      true /*enable_grad_clipping*/));
+      opt_graph_config_.enable_grad_norm_clip));
 
   return Status::OK();
 }
 
+static Status AddNcclAllReduceForGradientsWithGroups(
```

Comment:
Do we need #ifdef to avoid building this when we have MPI but do not have NCCL?

Reply:
refactored code related to the nccl directive and put a guard around this function.